rdkcentral/Dobby
# RDK-29822 - Port Sky's HttpProxy plugin into an RDK Plugin

### Description

Adds Sky's HttpProxy plugin (legacyPlugin style) as an rdkPlugin.

The HttpProxy plugin adds the ability to add `http_proxy` and `no_proxy` environment variables and proxy CA ceritificates to the container.

Disabled by default. Enabled with cmake flag `-DPLUGIN_HTTPPROXY`

### Test Procedure

To run test:
1. Fill in the `httpproxy` field in the `rdkPlugins` section of a container's config (see `rdkPlugins/HttpProxy/README.md` for instructions).
2. Launch container (make sure `postinstallhooksuccess` is not present)
3. Run `DobbyTool exec [container] cat /etc/ssl/certs/ca-certificates.crt`.

To confirm success:
1. Check that the environment variables are added to the container config as expected in the `process.env` section.
2. Check the container logs to confirm that the `proxyRootCACert` entry was properly prepended at the top of the output of reading `/etc/ssl/certs/ca-certificates.crt`.

### Type of Change

### Requires Bitbake Recipe changes?
